File: 2579-count-total-number-of-colored-cells/2579-count-total-number-of-colored-cells.cpp
class Solution {
public:
    long long coloredCells(int n) {
        
        vector<long long >dp(n+1,0);
        
        long long sum = 1,presum = 0;
        long long oddNum = 3;
        if(n == 1) return 1;
        
        for(int i = 2 ; i<=n ; i++){
            presum += oddNum - 2;
            sum += oddNum ;
            oddNum += 2;
            dp[i] = sum + presum ;
        }
        
        return dp[n];
    }
};

File: 2579-count-total-number-of-colored-cells/2579-count-total-number-of-colored-cells.js
/**
 * @param {number} n
 * @return {number}
 */
var coloredCells = function(n) {
    
    let dp = new Array(n+1).fill(0);
    
    let oddNum = 3;
    let presum = 0;
    let sum = 1;
    if(n == 1) return 1;
    for(let i = 2; i<=n; i++){
        presum += oddNum-2;
        sum += oddNum;
        oddNum+=2;
        dp[i] = presum + sum ;
    }
    return dp[n];
};